src/dotnet/UI.Blazor/Components/Modal/DiveInDialogFrame.razor

```diff
@@ -0,0 +1,93 @@
+﻿@using ActualChat.UI.Blazor.Pages.DiveInModalTestPage
+@namespace ActualChat.UI.Blazor.Components
+@implements IDiveInModalContext;
+
+@{
+    var pageContext = _pageInfo.Context;
+}
+
+<CascadingValue IsFixed="true" Value="@this">
+<CascadingValue Value="@pageContext">
+    @{
+        var title = pageContext.Title;
+        var cls = Class + " " + pageContext.Class;
+        var narrowViewSettings = _narrowViewSettingsBuilder.GetFrom(pageContext.ButtonInfos);
+    }
+    <DialogFrame
+        HasHeader="@HasHeader"
+        HasCloseButton="@HasCloseButton"
+        Title="@title"
+        ButtonsClass="@ButtonsClass"
+        Class="@cls"
+        NarrowViewSettings="@narrowViewSettings">
+        <Body>
+            @_pageInfo.Body
+        </Body>
+        <Buttons>
+            @if (pageContext.ButtonInfos != null) {
+                <DialogButtons ButtonInfos="@pageContext.ButtonInfos" Cancel="@OnClose" />
+            }
+        </Buttons>
+    </DialogFrame>
+</CascadingValue>
+</CascadingValue>
+
+@code {
+    private PageInfo _pageInfo = null!;
+    private DialogFrameNarrowViewSettingsBuilder _narrowViewSettingsBuilder = new();
+
+    [CascadingParameter] public Modal Modal { get; set; } = null!;
+
+    [Parameter] public string Class { get; set; } = "";
+    [Parameter] public string ButtonsClass { get; set; } = "";
+    [Parameter] public bool HasHeader { get; set; } = true;
+    [Parameter] public bool HasCloseButton { get; set; }
+    [Parameter, EditorRequired] public DiveInDialogPage StartPage { get; set; } = null!;
+
+    protected override void OnInitialized()
+        => _pageInfo = CreatePageInfo(StartPage);
+
+    private PageInfo CreatePageInfo(DiveInDialogPage page) {
+        var pageContext = new DiveInModalPageContext(this, page);
+        RenderFragment body = builder => RenderPage(page).Invoke(builder);
```

Comment:
local function instead of delegate variable?

Reply:
or even 
```csharp
return new PageInfo(pageContext, pageId, RenderPage(page));
```

Reply:
I used last approach.